core/java/iesi-core/src/test/java/io/metadew/iesi/script/execution/instruction/data/text/TextSubstringTest.java

```diff
@@ -61,4 +61,75 @@ void substringTwoArgumentsNegativeArguments() {
         assertEquals("ststring", textSubstring.generateOutput("teststring, -8"));
     }
 
+    @Test
+    void substringTextWithNewlines() {
+        TextSubstring textSubstring = new TextSubstring();
+        assertEquals("second line", textSubstring.generateOutput("first line\n" +
+                "second line\n" +
+                "third line\n" +
+                "fourth line\n" +
+                "fifth line\n" +
+                "sixth line\n" +
+                "seventh line\n" +
+                "eighth line\n, 11, 21"));
+
+    }
+
+    @Test
+    void substringTextWithNewlinesCross() {
+        TextSubstring textSubstring = new TextSubstring();
+        assertEquals(
+                "first line\n" +
+                "second line\n" + "third line", textSubstring.generateOutput("first line\n" +
+                "second line\n" +
+                "third line\n" +
+                "fourth line\n" +
+                "fifth line\n" +
+                "sixth line\n" +
+                "seventh line\n" +
+                "eighth line\n, 1, 31"));
+
+    }
+
+    @Test
+    void substringTextWithNewlinesMinValue() {
+        TextSubstring textSubstring = new TextSubstring();
+        assertEquals("first line", textSubstring.generateOutput("first line\n" +
+                "second line\n" +
+                "third line\n" +
+                "fourth line\n" +
+                "fifth line\n" +
+                "sixth line\n" +
+                "seventh line\n" +
+                "eighth line\n, 1, 10"));
+
+    }
+
+    @Test
+    void substringTextWithNewlinesLastValue() {
+        TextSubstring textSubstring = new TextSubstring();
+        assertEquals("eighth line", textSubstring.generateOutput("first line\n" +
+                "second line\n" +
+                "third line\n" +
+                "fourth line\n" +
+                "fifth line\n" +
+                "sixth line\n" +
+                "seventh line\n" +
+                "eighth line\n, 75, 85"));
+
+    }
+
+    @Test
+    void substringTextWithNewlinesTwoArguments() {
+        TextSubstring textSubstring = new TextSubstring();
+        assertEquals("eighth line\n", textSubstring.generateOutput("first line\n" +
+                "second line\n" +
+                "third line\n" +
+                "fourth line\n" +
+                "fifth line\n" +
+                "sixth line\n" +
+                "seventh line\n" +
+                "eighth line\n, 80"));
+
+    }
```

Comment:
formatting of code. In IntelliJ you can auto format the code. Normally SonarLint should also complain about this formatting. Is the plugin installed at your side?